sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -598,6 +598,23 @@ class DataFrameSuite extends QueryTest {
     testData.select($"*").show(1000)
   }
 
+  test("SPARK-7319 showString") {
+    assert(testData.select($"*").showString(1) === """+---+-----+
```

Comment:
it's pretty awkward to have this at the end. I'd just create a variable for the expected string so we don't have so much whitespace.